src/app/containers/Header/ScriptLink/index.test.jsx

```diff
@@ -0,0 +1,208 @@
+import { service as ukChinaServiceConfig } from '#lib/config/services/ukchina';
+import * as cookies from '#contexts/UserContext/cookies';
+import ScriptLinkContainer, { getVariantHref } from '.';
+import {
+  articlePath,
+  cpsAssetPagePath,
+  errorPagePath,
+  frontPagePath,
+  legacyAssetPagePath,
+} from '#app/routes/utils/regex';
+
+const setPreferredVariantCookieSpy = jest.spyOn(
+  cookies,
+  'setPreferredVariantCookie',
+);
+
+const userContextMock = {
+  setPreferredVariantCookie: cookies.setPreferredVariantCookie,
+};
+
+const requestContextMock = {
+  variant: 'lat',
+};
+
+const withRouter = (component, matchPath, path) => {
+  const Wrapper = ({ children }) => (
+    <MemoryRouter initialEntries={[path]}>
+      <Route path={matchPath}>{children}</Route>
+    </MemoryRouter>
+  );
+
+  Wrapper.propTypes = {
+    children: node.isRequired,
+  };
+
+  return {
+    ...render(component, { wrapper: Wrapper }),
+  };
+};
+
+/* eslint-disable react/prop-types */
+const ScriptLinkContainerWithContext = ({
+  serviceContext = serbianServiceConfig.lat,
+  requestContext = requestContextMock,
+}) => (
+  <ServiceContext.Provider value={serviceContext}>
+    <UserContext.Provider value={userContextMock}>
+      <RequestContext.Provider value={requestContext}>
+        <ScriptLinkContainer />
+      </RequestContext.Provider>
+    </UserContext.Provider>
+  </ServiceContext.Provider>
+);
+
+describe(`Script Link`, () => {
+  shouldMatchSnapshot(
+    'should render correctly',
+    <MemoryRouter initialEntries={['/serbian/lat']}>
+      <Route path={frontPagePath}>
+        <ScriptLinkContainerWithContext />
+      </Route>
+    </MemoryRouter>,
+  );
+
+  describe('assertions', () => {
+    const testCases = {
+      article: {
+        matchPath: articlePath,
+        path: '/serbian/articles/c805k05kr73o/lat',
+        variantPath: '/serbian/articles/c805k05kr73o/cyr',
+      },
+      cpsAssetPage: {
+        matchPath: cpsAssetPagePath,
+        path: '/serbian/lat/srbija-46748932',
+        variantPath: '/serbian/cyr/srbija-46748932',
+      },
+      errorPage: {
+        matchPath: errorPagePath,
+        path: '/serbian/404/lat',
+        variantPath: '/serbian/404/cyr',
+      },
+      frontPage: {
+        matchPath: frontPagePath,
+        path: '/serbian/lat',
+        variantPath: '/serbian/cyr',
+      },
+      legacyAssetPage: {
+        matchPath: legacyAssetPagePath,
+        path:
+          '/ukchina/trad/multimedia/2015/11/151120_video_100w_london_chinese_entrepreneurs',
+        variantPath:
+          '/ukchina/simp/multimedia/2015/11/151120_video_100w_london_chinese_entrepreneurs',
+        serviceContext: ukChinaServiceConfig.trad,
+        requestContext: { variant: 'trad' },
+        otherVariant: 'simp',
+      },
+    };
+
+    describe('canonical', () => {
+      Object.keys(testCases).forEach(testCase => {
+        it(`Script Link should contain link to other variant when on ${testCase}`, () => {
```

Comment:
We have this level of integration testing, so we should be fine with removing the Cypress tests above. cc @PriyaKR